File: software/examples/csharp/ExamplePixelMatrix.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change to your UID
	private static int WIDTH = 128;
	private static int HEIGHT = 64;

	private static void DrawMatrix(BrickletOLED128x64 oled, bool[][] pixels)
	{
		byte[][] pages = new byte[HEIGHT / 8][];

		for (int row = 0; row < HEIGHT / 8; row++)
		{
			pages[row] = new byte[WIDTH];

			for (int column = 0; column < WIDTH; column++)
			{
				pages[row][column] = 0;

				for (int bit = 0; bit < 8; bit++)
				{
					if (pixels[(row * 8) + bit][column])
					{
						pages[row][column] |= (byte)(1 << bit);
					}
				}
			}
		}

		oled.NewWindow(0, (byte)(WIDTH - 1), 0, (byte)(HEIGHT / 8 - 1));

		byte[] section = new byte[64];

		for (int row = 0; row < HEIGHT / 8; row++)
		{
			for (int column = 0; column < WIDTH; column += 64) {
				Array.Copy(pages[row], column, section, 0, 64);
				oled.Write(section);
			}
		}
	}

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletOLED128x64 oled = new BrickletOLED128x64(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Clear display
		oled.ClearDisplay();

		// Draw checkerboard pattern
		bool[][] pixels = new bool[HEIGHT][];

		for (int row = 0; row < HEIGHT; row++)
		{
			pixels[row] = new bool[WIDTH];

			for (int column = 0; column < WIDTH; column++)
			{
				pixels[row][column] = (row / 8) % 2 == (column / 8) % 2;
			}
		}

		DrawMatrix(oled, pixels);

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}

File: software/examples/perl/example_pixel_matrix.pl
#!/usr/bin/perl

use Tinkerforge::IPConnection;
use Tinkerforge::BrickletOLED128x64;

use constant HOST => 'localhost';
use constant PORT => 4223;
use constant UID => 'XYZ'; # Change to your UID
use constant WIDTH => 128;
use constant HEIGHT => 64;

sub draw_matrix
{
    my ($oled, $pixels_ref) = @_;
    my @pixels = @{$pixels_ref};
    my @pages = ();

    foreach my $row (0..&HEIGHT / 8 - 1) {
        $pages[$row] = ();

        foreach my $column (0..&WIDTH - 1) {
            $pages[$row][$column] = 0;

            foreach my $bit (0..7) {
                if ($pixels[($row * 8) + $bit][$column]) {
                    $pages[$row][$column] |= 1 << $bit;
                }
            }
        }
    }

    $oled->new_window(0, &WIDTH - 1, 0, &HEIGHT / 8 - 1);

    foreach my $row (0..&HEIGHT / 8 - 1) {
        for (my $column = 0; $column < &WIDTH; $column += 64) {
            my @section = @{$pages[$row]}[$column..$column + 64];

            $oled->write(\@section);
        }
    }
}

my $ipcon = Tinkerforge::IPConnection->new(); # Create IP connection
my $oled = Tinkerforge::BrickletOLED128x64->new(&UID, $ipcon); # Create device object

$ipcon->connect(&HOST, &PORT); # Connect to brickd
# Don't use device before ipcon is connected

# Clear display
$oled->clear_display();

# Draw checkerboard pattern
my @pixels = ();

foreach my $row (0..&HEIGHT - 1) {
    $pixels[$row] = ();

    foreach my $column (0..&WIDTH - 1) {
        $pixels[$row][$column] = (($row / 8) % 2) == (($column / 8) % 2);
    }
}

draw_matrix($oled, \@pixels);

print "Press key to exit\n";
<STDIN>;
$ipcon->disconnect();
